tests/test_commands.py

```diff
@@ -1472,3 +1638,152 @@ def verify_reporting_row(row, time_date):
     reporting_rows = Reporting.query.all()
     for row in reporting_rows:
         verify_reporting_row(row=row, time_date=first_time if row.id == 1 else second_time)
+
+
+def test_send_usage(client, cli_runner, capfd: LogCaptureFixture):
+    """Test that the email with the usage report is send"""
+    # Imports
+    from dds_web.database.models import Usage
+
+    # Get projects
+    projects = models.Project.query.filter(
+        models.Project.public_id.in_(
+            ["public_project_id", "second_public_project_id", "unit2testing"]
+        )
+    ).all()
+    project_1_unit_1 = next(p for p in projects if p.public_id == "public_project_id")
+    project_2_unit_1 = next(p for p in projects if p.public_id == "second_public_project_id")
+    project_1_unit_2 = next(p for p in projects if p.public_id == "unit2testing")
+
+    # Loop to populate usage table with fake entries across two years
+    january_2021 = datetime(2021, 1, 1)  # Start at Jan 2021
+    usage_list = []
+    for i in range(25):
+        time = january_2021 + relativedelta(months=i)
+        usage_1 = Usage(
+            project_id=project_1_unit_1.id,
+            usage=100,
+            time_collected=time,
+        )
+        usage_2 = Usage(
+            project_id=project_2_unit_1.id,
+            usage=100,
+            time_collected=time,
+        )
+        usage_3 = Usage(
+            project_id=project_1_unit_2.id,
+            usage=100,
+            time_collected=time,
+        )
+        usage_list.extend([usage_1, usage_2, usage_3])
+
+    db.session.add_all(usage_list)
+    db.session.commit()
+    # Fake data included from Jan 2021 to Jan 2023
+
+    def run_command_and_check_output(months_to_test, start_time):
+        # Run command
+        with mail.record_messages() as outbox:
+            with patch("dds_web.utils.current_time") as current_time_func:  # Mock current time
+                current_time_func.return_value = start_time
+                cli_runner.invoke(send_usage, ["--months", months_to_test])
+
+            # Verify output and sent email
+            assert len(outbox) == 1
+            assert (
+                "[SEND-USAGE CRONJOB] Usage records attached in the present mail"
+                in outbox[-1].subject
+            )
+            assert f"Here is the usage for the last {months_to_test} months." in outbox[-1].body
+
+            start_month = start_time.month
+            end_month = start_month - months_to_test
+            if end_month <= 0:
+                end_month += 12
```

Comment:
Just checking - is this needed? Feels like the dates should be able to handle substraction themselves, e.g. that `start_time-relativedelta(months_to_test)` should give the previous year.